include/onnxruntime/core/providers/coreml/coreml_provider_factory.h

```diff
@@ -28,8 +24,14 @@ enum COREMLFlags {
   // dynamic shapes. However, the performance may be negatively impacted if inputs have dynamic shapes.
   COREML_FLAG_ONLY_ALLOW_STATIC_INPUT_SHAPES = 0x008,
 
+  // Using CPU only in CoreML EP, this may decrease the perf but will provide
+  // reference output value without precision loss, which is useful for validation
+  COREML_FLAG_USE_CPU_ONLY = 0x010,
+  COREML_FLAG_USE_CPUAndGPU = 0x020,
+  COREML_FLAG_USE_CPUAndNeuralEngine = 0x040,
+  COREML_FLAG_USE_UnitAll = 0x080,
   // Create an MLProgram. By default it will create a NeuralNetwork model. Requires Core ML 5 or later.
-  COREML_FLAG_CREATE_MLPROGRAM = 0x010,
+  COREML_FLAG_CREATE_MLPROGRAM = 0x100,
```

Comment:
We shouldn't change any existing flag as that will break any current usage of the EP.

Could we maybe just add a new 'no ANE' flag if the problem we're trying to solve is that large data sizes run less efficiently if CoreML attempts to use the ANE?

Reply:
Yes, we could.

But the flag's value shouldn't be really matter given that we are using this enum variable. 
Please Correct me if I am wrong.

Reply:
Besides, 'no ANE' wouldn't always give the best performance on different Mac.

Reply:
even if users use the enum, if they have an older version of the header, their old enum value might now mean something else. that would be unexpected.